frontend/scripts/metrica.js

```diff
@@ -0,0 +1,231 @@
+            return device.blackberry() && !find('tablet');
+        };
+
+        device.blackberryTablet = () => {
+            return device.blackberry() && find('tablet');
+        };
+
+        device.windows = () => {
+            return find('windows');
+        };
+
+        device.windowsPhone = () => {
+            return device.windows() && find('phone');
+        };
+
+        device.windowsTablet = () => {
+            return device.windows() && (find('touch') && !device.windowsPhone());
+        };
+
+        device.fxos = () => {
+            return (find('(mobile;') || find('(tablet;')) && find('; rv:');
+        };
+
+        device.fxosPhone = () => {
+            return device.fxos() && find('mobile');
+        };
+
+        device.fxosTablet = () => {
+            return device.fxos() && find('tablet');
+        };
+
+        device.meego = () => {
+            return find('meego');
+        };
+
+        device.cordova = () => {
+            return window.cordova && location.protocol === 'file:';
+        };
+
+        device.nodeWebkit = () => {
+            return typeof window.process === 'object';
+        };
+
+        device.mobile = () => {
+            return device.androidPhone() ||
+                device.iphone() ||
+                device.ipod() ||
+                device.windowsPhone() ||
+                device.blackberryPhone() ||
+                device.fxosPhone() ||
+                device.meego();
+        };
+
+        device.tablet = () => {
+            return device.ipad() ||
+                device.androidTablet() ||
+                device.blackberryTablet() ||
+                device.windowsTablet() ||
+                device.fxosTablet();
+        };
+
+        device.desktop = () => {
+            return !device.tablet() && !device.mobile();
+        };
+
+        find = (needle) => {return userAgent.indexOf(needle) !== -1};
+
+        return device;
+    };
+
+    const getUserAgent = () => {
+        return window.navigator.userAgent;
+    };
+
+    const setToken = (token) => {
+        localStorage.setItem('visitor_token', token);
+    };
+
+    const getToken = () => {
+        if(localStorage.getItem('visitor_token') !== null){
+            return localStorage.getItem("visitor_token");
+        }
+        return null;
+    };
+
+    const getPage = () => {
+        return window.location.href;
+    };
+
+    const getTitle = () => {
+        return document.title;
+    };
+
+    const getLanguage = () => {
+        return navigator.language || navigator.userLanguage;
+    };
+
+    const getDevice = () => {
+        let device = 'unknown';
+        if(getUserDevice().desktop()) {
+            device = 'desktop'
+        } else if(getUserDevice().tablet()) {
+            device = 'tablet'
+        } else if(getUserDevice().mobile()) {
+            device = 'mobile'
+        }
+        return device;
+    };
+
+    const getResolutionWith = () => {
+        return screen.width;
```

Comment:
https://caniuse.com/#search=screen

property screen is not supported by some browsers, be careful